### PR TITLE
MAINT: remove unneeded image_mode references

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
This PR cleans up the codebase by refactoring references to `request.mode.image_mode`. The goal is to remove all references to image_mode and ultimately remove it from the codebase. This is a first step towards refactoring the request object into what we discussed in #798 .

### CODE REVIEW

**Comment by FirefoxMetzger** [comment]
This PR is a bit old, but it is RTR as well.

For this one, I will not merge it directly, but instead wait for @almarklein or someone else to look this over before I merge.

File: imageio/core/legacy_plugin_wrapper.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def legacy_get_writer(self, **kwargs):
         self._request._kwargs = kwargs
         return self._format.get_writer(self._request)
 
-    def write(self, ndimage, **kwargs):
+    def write(self, ndimage, is_batch=None, **kwargs):
```
Comment (almarklein) [inline imageio/core/legacy_plugin_wrapper.py:165]: Needs documentation too?

**Review by almarklein — COMMENTED** [review]
One question. Previously the user could specify single-image, multi-image or auto. This is still the same, except now auto is resolved in `write()`. How would this work for `read`? Because there we want to resolve auto in the plugins, e.g. to check tiff metadata to determine whether its an image series or volume.

File: imageio/core/legacy_plugin_wrapper.py [hunk 2]
```diff
@@ -179,41 +179,34 @@ def write(self, ndimage, **kwargs):
...
+            is_batch = True
+        elif ndimage.ndim == 2:
+            is_batch = False
+        elif ndimage.ndim == 3 and ndimage.shape[2] < 5:
+            is_batch = False
+        else:
+            is_batch = True
+
+        if not is_batch:
+            ndimage = np.asarray(ndimage)[None, ...]
```
[the remark this answers is not in the document]
Comment (FirefoxMetzger) [reply imageio/core/legacy_plugin_wrapper.py:195]: If the provided ndimage is not flagged as a batch of images we prepend a unitary dimension, i.e., we turn it into a "batch" containing a single image. This way we can treat both cases in the same way and combine code paths.

File: imageio/core/legacy_plugin_wrapper.py (hunk 1 above)
Comment (FirefoxMetzger) [reply imageio/core/legacy_plugin_wrapper.py]: Yep. I added the docs :)

**Pull request merged**